### PR TITLE
add yarn, nodejs; install katex for docs

### PR DESCRIPTION
**Pull request by mattip**
- add yarn and nodejs to the conda environment
- add katex to env/bin for doc building using yarn
~- prefer USE_XNNPACK=1, maybe should depend on the commit that broke USE_XNNPACK=0~
- add `-f` to git submodule update command
- add another `d` to git clean to remove empty directories
- reduce the number of lines in EndOfMessage so more info is displayed

### CODE REVIEW

**Comment by mattip** [comment]
@pearu rebased PR

File: working-envs/activate-pytorch-dev.sh [hunk 1]
```diff
@@ -116,6 +117,14 @@ else
     export USE_NCCL=0
 fi
 
+if [[ -z "$(command -v katex)" ] -a [ -x "$(command -v yarn)"]]
+then
+    # Try to install katex to the bin/ directory of the conda environment
+    yarn global add katex $(dirname $(dirname `which python`))
```
Comment (pearu) [inline working-envs/activate-pytorch-dev.sh:123]: This command should read `yarn global add latex --prefix $CONDA_PREFIX`

**Review by pearu — CHANGES_REQUESTED** [review]
Thanks for this PR!
Could you take a look at
https://github.com/Quansight/pearu-sandbox/commit/a8c166f05a4062124831bc9855c143125852266b
and copy the katex detection/installation logic here?
I rewrote it because it didn't work out of the box and later discovered that the `--prefix` option was missing in the yarn command line.
Also, I disabled yarn command to be automatically executed because the activate script should just set the environment and give hints, user will choose whether to follow these or not.

**Comment by pearu** [comment]
You can ignore `USE_ENV` stuff, I can add that later.

**Comment by mattip** [comment]
LGTM. The build doc command is `cd docs; python -m pip install requirements.txt; PYTHONPATH=.. make html`, but that command is meant to be run after `python setup.py develop`.

**Comment by pearu** [comment]
Closing via PR #15

**Pull request closed without merging**